app/questionnaire/value_source_resolver.py

```diff
@@ -114,8 +117,37 @@ def _resolve_list_value_source(self, value_source: dict) -> Union[int, str, list
 
         return list(list_model)
 
+    def _resolve_calculated_summary_value_source(
+        self, value_source: Mapping
+    ) -> IntOrDecimal:
+        """Calculates the value for the 'calculation' used by the provided Calculated Summary.
+
+        The caller is responsible for ensuring the provided Calculated Summary and its answers are on the path.
+        """
+        calculated_summary_block: Mapping[str, Any] = self.schema.get_block(value_source["identifier"])  # type: ignore
+        calculation = calculated_summary_block["calculation"]
+        answers_to_calculate = calculation["answers_to_calculate"]
+        operator = self.get_calculation_operator(calculation["calculation_type"])
+        list_item_id = self._resolve_list_item_id_for_value_source(value_source)
+
+        values = [
+            self._get_answer_value(answer_id=answer_id, list_item_id=list_item_id)
+            for answer_id in answers_to_calculate
+        ]
+        total: IntOrDecimal = operator([value for value in values if value])  # type: ignore
+        return total
```

Comment:
I would shorten it even more but it's a question of personal preference.
```suggestion
        calculated_summary_block: Mapping[str, Any] = self.schema.get_block(value_source["identifier"])  # type: ignore
        calculation = calculated_summary_block["calculation"]
        operator = self.get_calculation_operator(calculation["calculation_type"])
        list_item_id = self._resolve_list_item_id_for_value_source(value_source)

        values = [
            self._get_answer_value(answer_id=answer_id, list_item_id=list_item_id)
            for answer_id in calculation["answers_to_calculate"]
        ]
        return operator([value for value in values if value])
```

Reply:
Since we already have a docstring we don't need to assign a name to the returned value?

Reply:
It was initially for type hint reasons i.e `total: IntOrDecimal` but have changed anyway, as the return is already `IntOrDecimal` so `type: ignore` for the return statement would be unaffected.